effectsmanager.h

```diff
@@ -190,5 +216,86 @@ class EffectsManager : public IEffectsManager
     {
         return _currentEffectIndex >= 0 && _currentEffectIndex < static_cast<int>(_effects.size());
     }
+
+    friend void to_json(nlohmann::json& j, const EffectsManager& manager);
+    friend void from_json(const nlohmann::json& j, EffectsManager& manager);
 };
 
+
+inline void to_json(nlohmann::json& j, const ILEDEffect& effect) 
```

Comment:
This type of repetitive code (and the corresponding code in the from_json() below it) make my fingers itch to "macro" it, but... I'll refrain. For now. :)

Reply:
Consider snapping yourself with a rubber band each time you write a macro, it'll reduce the urge :-)

Reply:
That's about as sensible as suggesting I punch myself in the face every time I empty the dishwasher. So no, I don't think I will. ;)